Code/Framework/AzToolsFramework/AzToolsFramework/UI/DocumentPropertyEditor/DocumentPropertyEditorSettings.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzToolsFramework/UI/DocumentPropertyEditor/SettingsRegistrar.h>
+
+namespace AzToolsFramework
+{
+    //! This serializable class stores and loads the DocumentPropertyEditor settings such as tree node expansion state.
+    class DocumentPropertyEditorSettings
+    {
+    public:
+        AZ_RTTI(DocumentPropertyEditorSettings, "{7DECB0A1-A1AB-41B2-B31F-E52D3C3014A6}");
+
+        using ExpanderStateMap = AZStd::unordered_map<AZStd::string, bool>;
+        using CleanExpanderStateCallback = AZStd::function<bool(ExpanderStateMap&)>;
+
+        // Default ctor is required by SerializeContext but is not intended for use otherwise
+        DocumentPropertyEditorSettings() = default;
+        DocumentPropertyEditorSettings(const AZStd::string& settingsRegistryKey, const AZStd::string& propertyEditorName);
+
+        virtual ~DocumentPropertyEditorSettings();
+
+        static void Reflect(AZ::ReflectContext* context)
+        {
+            AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context);
+            if (serializeContext)
+            {
+                serializeContext->Class<DocumentPropertyEditorSettings>()->Version(0)->Field(
+                    "ExpandedElements", &DocumentPropertyEditorSettings::m_expandedElementStates);
+            }
+        }
+
+        void SetExpanderStateForRow(const AZ::Dom::Path& rowPath, bool isExpanded);
+        bool GetExpanderStateForRow(const AZ::Dom::Path& rowPath);
```

Comment:
Nitpick: Could make this method `const`

```suggestion
        bool GetExpanderStateForRow(const AZ::Dom::Path& rowPath) const;
```

Reply:
Done.